quinn-proto/src/connection/mod.rs

```diff
@@ -673,18 +684,22 @@ impl Connection {
         self.endpoint_events
             .push_back(EndpointEventInner::RetireResetToken(path_id));
 
-        self.abandoned_paths.insert(path_id);
+        let pto = self.pto_max_path(SpaceId::Data);
+
+        // We record the time at which receiving data on this path generates a transport error.
+        self.abandoned_paths.insert(path_id, now + 3 * pto);
```

Comment:
I think this is very strange place to put that timeout in the current code.  Because this timeout is not relevant for most of the lifetime that `abandoned_paths` exists for.

It would make a lot more sense to place the 3 * max PTO in the `PathData` which is still retained here and is only removed when the timer below expires. That has exactly the right lifetime for this. (It also makes a lot of the unrelated changes wrt to abandoned_path go away.)